mempool/src/shared_mempool/priority.rs

```diff
@@ -229,8 +402,27 @@ impl PrioritizedPeersState {
                 .all(|(_, metadata)| get_peer_ping_latency(metadata).is_some());
         }
 
+        // Divide the sender buckets amongst the top peers
+        self.update_sender_bucket_for_peers(
```

Comment:
When num_sender_buckets = 1, do we still do this update? Maybe it makes sense to skip this when =1, so we have a config that is gating the new behavior in case there are issues?

Reply:
Yeah, we still to update what that 1 sender bucket is assigned to. But the logic of the code effectively is same as before. We assign the top peer as primary and the second top peer as Failover to that 1 sender bucket.